atlasdb-config/src/main/java/com/palantir/atlasdb/http/NotInitializedExceptionMapper.java

```diff
@@ -15,18 +15,34 @@
  */
 package com.palantir.atlasdb.http;
 
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
 
+import com.palantir.conjure.java.api.errors.QosException;
 import com.palantir.exception.NotInitializedException;
 
 /**
- * Maps a {@link NotInitializedException} to a 503 status code.
+ * Converts {@link NotInitializedException}s into appropriate status responses, depending on the user's
+ * {@link AtlasDbHttpProtocolVersion}. The intention is that clients may retry on an arbitrary node of the service,
+ * and they should backoff, as that node of the cluster is still starting up.
+ *
+ * This is a 503 without a Retry-After header and with a message body corresponding to {@link NotInitializedException}
+ * in {@link AtlasDbHttpProtocolVersion#LEGACY_OR_UNKNOWN}.
  */
 public class NotInitializedExceptionMapper implements ExceptionMapper<NotInitializedException> {
+    @Context
+    private HttpHeaders httpHeaders;
+
+    private static final HttpProtocolAwareExceptionTranslator<NotInitializedException> translator = new
+            HttpProtocolAwareExceptionTranslator<>(
+            AtlasDbHttpProtocolHandler.LambdaHandler.of(
```

Comment:
nit: torn between using the builder of `LambdaHandler` to make it a bit more obvious as to what each parameter means e.g.:

```java
AtlasDbHttpProtocolHandler.LambdaHandler.builder()
  .legacyBehaviour(...)
  .version2Behaviour(...)
  .build();
```

We basically have a visitor right?

Reply:
This makes sense - though the lambda handler shouldn't exist any more as we move to an abstract class.